kotlin-analysis-api/src/main/kotlin/com/google/devtools/ksp/impl/ResolverAAImpl.kt

```diff
@@ -134,8 +142,29 @@ class ResolverAAImpl(
         }
     }
 
+    @KspExperimental
     override fun getDeclarationsFromPackage(packageName: String): Sequence<KSDeclaration> {
-        TODO("Not yet implemented")
+        return analyze {
+            val packageNames = packageName.split(".")
+            var packages = listOf(analysisSession.ROOT_PACKAGE_SYMBOL)
+            for (curName in packageNames) {
+                packages = packages
+                    .flatMap { it.getPackageScope().getPackageSymbols { it.asString() == curName } }
+                    .distinct()
+            }
+            packages.flatMap {
+                it.getPackageScope().getAllSymbols().distinct().mapNotNull { symbol ->
```

Comment:
Why does it need `distinct()`?

Reply:
AA returns duplicate package symbols for reasons I don't know. Probably need a fix in upstream but put distinct() here for now.